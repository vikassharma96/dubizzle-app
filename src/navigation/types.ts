import type {NativeStackScreenProps} from '@react-navigation/native-stack';

export type RootStackParamList = {
  Home: undefined;
  Info: {uri: string};
};

export type HomeScreenProps = NativeStackScreenProps<
  RootStackParamList,
  'Home'
>;
export type InfoScreenProps = NativeStackScreenProps<
  RootStackParamList,
  'Info'
>;
