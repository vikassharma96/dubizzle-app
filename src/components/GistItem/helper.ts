import {IGist} from '../../types/gist.types';
import {IGistMeta} from '../../types/gistmeta.types';

const getGistMeta = (gist: IGist): IGistMeta => {
  const {
    files = {},
    html_url,
    forks_url,
    comments_url,
    comments = 0,
    description = '',
    created_at,
    updated_at,
  } = gist;
  return {
    files,
    html_url,
    forks_url,
    comments_url,
    comments,
    description,
    created_at,
    updated_at,
  };
};

export {getGistMeta};
