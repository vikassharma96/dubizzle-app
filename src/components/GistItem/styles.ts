import {StyleSheet} from 'react-native';
import colors from '../../styles/colors';
import {fontFamily, fontSizes} from '../../styles/fonts';

const styles = StyleSheet.create({
  container: {
    width: '100%',
  },
  description: {
    marginBottom: '2%',
    fontFamily: fontFamily.regular,
  },
  row: {
    width: '100%',
    flexDirection: 'row',
  },
  bottom: {
    marginBottom: '2%',
    marginStart: '3%',
  },
  files: {
    marginEnd: '1%',
    alignItems: 'center',
  },
  fileText: {
    marginStart: '1.5%',
    color: colors.blue,
    fontSize: fontSizes.fontSize_12,
    fontFamily: fontFamily.regular,
  },
});

export default styles;
