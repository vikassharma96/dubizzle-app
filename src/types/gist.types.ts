type IGist = {
  comments: number;
  comments_url: string;
  commits_url: string;
  created_at: string;
  description: string;
  files: any;
  forks_url: string;
  git_pull_url: string;
  git_push_url: string;
  html_url: string;
  id: string;
  node_id: string;
  owner: IOwner;
  public: boolean;
  truncated: boolean;
  updated_at: string;
  url: string;
  user?: null;
};

type IOwner = {
  avatar_url: string;
  events_url: string;
  followers_url: string;
  following_url: string;
  gists_url: string;
  gravatar_id: string;
  html_url: string;
  id: number;
  login: string;
  node_id: string;
  organizations_url: string;
  received_events_url: string;
  repos_url: string;
  site_admin: boolean;
  starred_url: string;
  subscriptions_url: string;
  type: string;
  url: string;
};

export type {IGist, IOwner};
