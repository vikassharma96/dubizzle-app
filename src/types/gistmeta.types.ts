type IGistMeta = {
  files: object;
  html_url: string;
  forks_url: string;
  comments_url: string;
  comments: number;
  description: string;
  updated_at: string;
  created_at: string;
};

export type {IGistMeta};
