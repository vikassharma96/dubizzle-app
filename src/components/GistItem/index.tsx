import React, {memo} from 'react';
import {TouchableOpacity} from 'react-native';
import GistHeader from '../GistHeader';
import Icon from 'react-native-vector-icons/Octicons';
import {ScrollView} from 'react-native';
import {IGist} from '../../types/gist.types';
import {IGistMeta} from '../../types/gistmeta.types';
import AppText from '../AppText';
import {getGistMeta} from './helper';
import styles from './styles';
import colors from '../../styles/colors';
import {Routes} from '../../navigation/routes';
import {useNavigation} from '@react-navigation/native';

type IProps = {
  gist: IGist;
};

const GistItem = ({gist}: IProps) => {
  const navigation = useNavigation();
  // owner info of the gist
  const {owner: gistOwner, files} = gist;
  // meta data of the gist
  const gistMeta: IGistMeta = getGistMeta(gist);
  // retrieve fileNames
  const fileNames = Object.keys(files);

  const onOpenUrl = (url: string) => {
    navigation.navigate(Routes.INFO, {
      uri: url,
    });
  };

  return (
    <>
      <GistHeader gistOwner={gistOwner} gistMeta={gistMeta} />
      {gistMeta?.description && (
        <AppText style={styles.description}>{gistMeta?.description}</AppText>
      )}
      <ScrollView
        horizontal
        contentContainerStyle={[styles.row, styles.bottom]}
        showsHorizontalScrollIndicator={false}>
        {fileNames.map((file: any) => {
          return (
            <TouchableOpacity
              style={[styles.row, styles.files]}
              key={file}
              onPress={() => onOpenUrl(files[file].raw_url)}>
              <Icon name={'file'} size={15} color={colors.blue} />
              <AppText style={styles.fileText}>{file}</AppText>
            </TouchableOpacity>
          );
        })}
      </ScrollView>
    </>
  );
};

export default memo(GistItem);
