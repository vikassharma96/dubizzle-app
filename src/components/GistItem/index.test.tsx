import React from 'react';
import renderer from 'react-test-renderer';
import GistItem from '.';
import {IGist, IOwner} from '../../types/gist.types';
import {IGistMeta} from '../../types/gistmeta.types';

describe('@components/GistItem', () => {
  const mockGistOwner = {
    login: 'vikassharma96',
    id: 51054221,
    node_id: 'MDQ6VXNlcjUxMDU0MjIx',
    avatar_url: 'https://avatars.githubusercontent.com/u/51054221?v=4',
    gravatar_id: '',
    url: 'https://api.github.com/users/vikassharma96',
    html_url: 'https://github.com/vikassharma96',
    followers_url: 'https://api.github.com/users/vikassharma96/followers',
    following_url:
      'https://api.github.com/users/vikassharma96/following{/other_user}',
    gists_url: 'https://api.github.com/users/vikassharma96/gists{/gist_id}',
    starred_url:
      'https://api.github.com/users/vikassharma96/starred{/owner}{/repo}',
    subscriptions_url:
      'https://api.github.com/users/vikassharma96/subscriptions',
    organizations_url: 'https://api.github.com/users/vikassharma96/orgs',
    repos_url: 'https://api.github.com/users/vikassharma96/repos',
    events_url: 'https://api.github.com/users/vikassharma96/events{/privacy}',
    received_events_url:
      'https://api.github.com/users/vikassharma96/received_events',
    type: 'User',
    site_admin: false,
  } as IOwner;
  const mockGistMeta = {
    files: {
      'pagination.js': {
        filename: 'pagination.js',
        type: 'application/javascript',
        language: 'JavaScript',
        raw_url:
          'https://gist.githubusercontent.com/vikassharma96/6ba7b96eaf7c7ea509f35d8451eed87e/raw/ad9d448678b01d33c6ff147b611b3bfcff816195/pagination.js',
        size: 2414,
      },
    },
    html_url:
      'https://gist.github.com/vikassharma96/6ba7b96eaf7c7ea509f35d8451eed87e',
    forks_url:
      'https://api.github.com/gists/6ba7b96eaf7c7ea509f35d8451eed87e/forks',
    comments_url:
      'https://api.github.com/gists/6ba7b96eaf7c7ea509f35d8451eed87e/comments',
    comments: 0,
    created_at: '2020-08-21T01:52:25Z',
    updated_at: '2020-08-21T01:52:26Z',
    description: 'Pagination in react native',
  } as IGistMeta;

  const mockGist = {
    owner: mockGistOwner,
    ...mockGistMeta,
  } as IGist;

  const navigate = jest.fn();

  const component = <GistItem gist={mockGist} navigation={{navigate}} />;

  it('should match snapshots', () => {
    const componentTree = renderer.create(component).toJSON();
    expect(componentTree).toMatchSnapshot();
  });
});
