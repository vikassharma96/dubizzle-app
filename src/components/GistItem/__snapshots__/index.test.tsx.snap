// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@components/GistItem should match snapshots 1`] = `
[
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": "2%",
        "marginEnd": "2%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          {
            "uri": "https://avatars.githubusercontent.com/u/51054221?v=4",
          }
        }
        style={
          {
            "borderRadius": 15,
            "height": 30,
            "width": 30,
          }
        }
      />
      <Text
        style={
          [
            {
              "fontFamily": "Poppins-Medium",
              "fontSize": 12,
            },
            {
              "color": "#000",
            },
            {
              "alignSelf": "center",
              "color": "#0366d6",
              "fontFamily": "Poppins-Medium",
              "fontSize": 12,
              "marginStart": "5%",
            },
          ]
        }
      >
        vikassharma96
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#0366d6",
                "fontSize": 16,
              },
              undefined,
              {
                "fontFamily": "Octicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "Poppins-Medium",
                "fontSize": 12,
              },
              {
                "color": "#000",
              },
              {
                "color": "#0366d6",
                "fontSize": 11,
                "paddingEnd": "3%",
                "paddingStart": "1.5%",
              },
            ]
          }
        >
          1 File
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#0366d6",
                "fontSize": 15,
              },
              {
                "paddingEnd": "4%",
              },
              {
                "fontFamily": "Octicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#0366d6",
                "fontSize": 15,
              },
              {
                "paddingEnd": "4%",
              },
              {
                "fontFamily": "Octicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#0366d6",
                "fontSize": 15,
              },
              undefined,
              {
                "fontFamily": "Octicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>,
  <Text
    style={
      [
        {
          "fontFamily": "Poppins-Medium",
          "fontSize": 12,
        },
        {
          "color": "#000",
        },
        {
          "color": "#6B6D6E",
          "fontFamily": "Poppins-Regular",
          "fontSize": 10,
          "marginBottom": "2%",
        },
      ]
    }
  >
    Created at: 8/21/2020  Last updated: 8/21/2020
  </Text>,
  <Text
    style={
      [
        {
          "fontFamily": "Poppins-Medium",
          "fontSize": 12,
        },
        {
          "color": "#000",
        },
        {
          "fontFamily": "Poppins-Regular",
          "marginBottom": "2%",
        },
      ]
    }
  >
    Pagination in react native
  </Text>,
  <RCTScrollView
    contentContainerStyle={
      [
        {
          "flexDirection": "row",
          "width": "100%",
        },
        {
          "marginBottom": "2%",
          "marginStart": "3%",
        },
      ]
    }
    horizontal={true}
    showsHorizontalScrollIndicator={false}
  >
    <View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginEnd": "1%",
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#0366d6",
                "fontSize": 15,
              },
              undefined,
              {
                "fontFamily": "Octicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "Poppins-Medium",
                "fontSize": 12,
              },
              {
                "color": "#000",
              },
              {
                "color": "#0366d6",
                "fontFamily": "Poppins-Regular",
                "fontSize": 12,
                "marginStart": "1.5%",
              },
            ]
          }
        >
          pagination.js
        </Text>
      </View>
    </View>
  </RCTScrollView>,
]
`;
